source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc

```diff
@@ -317,11 +317,33 @@ void RawHttpClientImpl::onBeforeFinalizeUpstreamSpan(
 ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
   const uint64_t status_code = Http::Utility::getResponseStatus(message->headers());
 
-  // Set an error status if the call to the authorization server returns any of the 5xx HTTP error
-  // codes. A Forbidden response is sent to the client if the filter has not been configured with
-  // failure_mode_allow.
   if (Http::CodeUtility::is5xx(status_code)) {
-    return std::make_unique<Response>(errorResponse());
+    // Set an error status if the call to the authorization server returns any of the 5xx HTTP error
+    // codes. A Forbidden response is sent to the client if the filter has not been configured with
+    // failure_mode_allow.
+    if (!Runtime::runtimeFeatureEnabled(
+      "envoy.reloadable_features.ext_authz_http_service_5xx_is_denied_instead_of_error")) {    
+      SuccessResponse error{message->headers(),
+                            config_->clientHeaderMatchers(),
+                            config_->upstreamHeaderToAppendMatchers(),
+                            config_->clientHeaderOnSuccessMatchers(),
+                            config_->dynamicMetadataMatchers(),
+                            Response{CheckStatus::Error,
+                                      Http::HeaderVector{},
+                                      Http::HeaderVector{},
+                                      Http::HeaderVector{},
+                                      Http::HeaderVector{},
+                                      Http::HeaderVector{},
+                                      {{}},
+                                      Http::Utility::QueryParamsVector{},
+                                      {},
+                                      message->bodyAsString(),
+                                      static_cast<Http::Code>(status_code),
+                                      ProtobufWkt::Struct{}}};
+      return std::move(error.response_);
```

Comment:
I think we don't need this since it will be handled by line 397?

Reply:
Hence the following patch is enough:

```diff
--- a/source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc
+++ b/source/extensions/filters/common/ext_authz/ext_authz_http_impl.cc
@@ -317,11 +317,14 @@ void RawHttpClientImpl::onBeforeFinalizeUpstreamSpan(
 ResponsePtr RawHttpClientImpl::toResponse(Http::ResponseMessagePtr message) {
   const uint64_t status_code = Http::Utility::getResponseStatus(message->headers());
 
-  // Set an error status if the call to the authorization server returns any of the 5xx HTTP error
-  // codes. A Forbidden response is sent to the client if the filter has not been configured with
-  // failure_mode_allow.
-  if (Http::CodeUtility::is5xx(status_code)) {
-    return std::make_unique<Response>(errorResponse());
+  if (!Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.ext_authz_http_service_5xx_is_denied")) {
+    // Set an error status if the call to the authorization server returns any of the 5xx HTTP error
+    // codes. A Forbidden response is sent to the client if the filter has not been configured with
+    // failure_mode_allow.
+    if (Http::CodeUtility::is5xx(status_code)) {
+      return std::make_unique<Response>(errorResponse());
+    }
   }
 
   // Extract headers-to-remove from the storage header coming from the
```

Reply:
ah oke mas. I initially thought the response should be `CheckStatus::Error` instead of `CheckStatus::Denied`. If it is denied, it makes sense that we don't need to do anything on the `http/ext_authz/ext_authz.cc`. I will update accordingly.